### PR TITLE
Implement deposit_authorized RPC and tests

### LINKED ISSUE #727: Implement `deposit_authorized` RPC handler
https://xrpl.org/deposit_authorized.html

### PR DESCRIPTION
**Pull request by godexsoft**
Fixes #727

### CODE REVIEW

File: src/rpc/handlers/DepositAuthorized.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    auto const lgrInfo = std::get<ripple::LedgerInfo>(lgrInfoOrStatus);
+    auto const sourceAccountID = accountFromStringStrict(input.sourceAccount);
+    auto const destinationAccountID = accountFromStringStrict(input.destinationAccount);
+
+    auto const srcAccountLedgerObject =
+        sharedPtrBackend_->fetchLedgerObject(ripple::keylet::account(*sourceAccountID).key, lgrInfo.seq, ctx.yield);
+
+    if (!srcAccountLedgerObject)
+        return Error{Status{RippledError::rpcACT_NOT_FOUND, "source_accountNotFound"}};
```
Comment (cindyyan317) [inline src/rpc/handlers/DepositAuthorized.cpp]: error code is wrong. should return rpcSRC_ACT_NOT_FOUND

File: src/rpc/handlers/DepositAuthorized.cpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    auto const dstKeylet = ripple::keylet::account(*destinationAccountID).key;
+    auto const dstAccountLedgerObject = sharedPtrBackend_->fetchLedgerObject(dstKeylet, lgrInfo.seq, ctx.yield);
+
+    if (!dstAccountLedgerObject)
+        return Error{Status{RippledError::rpcACT_NOT_FOUND, "destination_accountNotFound"}};
```
Comment (cindyyan317) [inline src/rpc/handlers/DepositAuthorized.cpp]: should return rpcDST_ACT_NOT_FOUND

**Review by cindyyan317 — CHANGES_REQUESTED** [review]
Send back because the error code mismatch

File: src/rpc/handlers/DepositAuthorized.cpp (hunk 1 above)
Comment (godexsoft) [reply src/rpc/handlers/DepositAuthorized.cpp]: Thanks for spotting this. Fixed now 👍

**Review by cindyyan317 — APPROVED** [review]
GOGOGO

**Pull request merged**